forum/projects/views.py

```diff
@@ -1,3 +1,16 @@
-from django.shortcuts import render
+from rest_framework.viewsets import ModelViewSet
+from rest_framework.permissions import IsAuthenticated
+from .models import Project
+from .serializers import ProjectSerializer
+from .permissions import IsOwnerOrReadOnly
 
-# Create your views here.
+
+class ProjectViewSet(ModelViewSet):
+    permission_classes = [IsAuthenticated, IsOwnerOrReadOnly]
+    queryset = Project.objects.all()
+    serializer_class = ProjectSerializer
+
+    def perform_create(self, serializer):
+        """Automatically assigns project to correct startup based on user's token"""
+        startup = self.request.user.startup_profile
+        serializer.save(startup=startup)
```

Comment:
Use `select_related` or `prefetch_related` to optimize database queries when accessing related objects like `startup_profile` and `description`.
```
def get_queryset(self):
    return Project.objects.filter(startup=self.request.user.startup_profile).select_related('startup')
```

Reply:
queryset in view was optimized using `select_related`